Production/govuk_ios/Repositories/TopicsRepository.swift

```diff
@@ -0,0 +1,88 @@
+import Foundation
+import CoreData
+
+protocol TopicsRepositoryInterface {
+    func saveTopicsList(_ topicResponses: [TopicResponseItem])
+    func fetchFavoriteTopics() -> [Topic]
+    func fetchAllTopics() -> [Topic]
+    func saveChanges()
+}
+
+struct TopicsRepository: TopicsRepositoryInterface {
+    private let coreData: CoreDataRepositoryInterface
+
+    init(coreData: CoreDataRepositoryInterface) {
+        self.coreData = coreData
+    }
+
+    func saveTopicsList(_ topicResponses: [TopicResponseItem]) {
+        let context = coreData.backgroundContext
+        let isFirstLaunch = fetchAllTopics().count == 0
+        topicResponses.forEach { topicResponse in
+            if !topicExists(for: topicResponse,
+                            in: context) {
+                createTopic(for: topicResponse,
+                            in: context,
+                            isFavorite: isFirstLaunch)
+            }
+        }
+        try? context.save()
+    }
+
+    func saveChanges() {
+        try? coreData.viewContext.save()
+    }
+
+    func fetchFavoriteTopics() -> [Topic] {
+        fetch(
+            predicate: .init(format: "isFavorite = true"),
+            context: coreData.viewContext
+        ).fetchedObjects ?? []
+    }
+
+    func fetchAllTopics() -> [Topic] {
+        fetch(
+            predicate: nil,
+            context: coreData.viewContext
+        ).fetchedObjects ?? []
+    }
+
+    private func fetchTopic(ref: String,
+                            context: NSManagedObjectContext
+    ) -> Topic? {
```

Comment:
Is this because of line length? Can you put the `)` on line 51 and start the new line with `->`, please?

Reply:
Not length, not sure why that got in there.  Now formattted like other functions in the file